sqlglot/parser.py

```diff
@@ -3060,11 +3060,7 @@ def _parse_column_ops(self, this: exp.Expression) -> exp.Expression:
                     else exp.Literal.string(value)
                 )
             else:
-                field = (
-                    self._parse_star()
-                    or self._parse_function(anonymous=True)
-                    or self._parse_id_var()
-                )
+                field = self._parse_field(anonymous_func=True)
```

Comment:
This was changed because in `<something>.1` we used to parse `1` as an identifier, so with this PR it was force-quoted and several tests broke (that I guess wanted `1` to be treated as an index into some nested type instead).